app/scripts/background.js

```diff
@@ -313,6 +313,7 @@ function setupController (initState, initLangCode) {
   //
   extension.runtime.onConnect.addListener(connectRemote)
   extension.runtime.onConnectExternal.addListener(connectExternal)
+  extension.runtime.onMessage.addListener(controller.onMessage.bind(controller))
```

Comment:
Does this only listen for events emitted by explicit calls to `extension.runtime.sendMessage()`?

Reply:
Yes, I believe so. Either that or `extension.tabs.sendMessage`, according to [this](https://developer.chrome.com/extensions/runtime#event-onMessage) and [this](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage).